index.js

```diff
@@ -109,6 +111,10 @@ module.exports = {
         return;
       }
 
+      if (policyObject) {
```

Comment:
For clarity, how about checking for `options.environment === 'test'` here too

Reply:
Actually this needs to be applied also for `options.environment === 'development'` cause tests may be executed by `http://localhost:4200/tests`. In that case `option.environment` is true. So would end up checking the request target also...

Reply:
@jelhan Alright, I haven't used `http://localhost:4200/tests` and I believe you, but it sounds weird that they aren't running under the test environment. Is that by design?

Anyway, perhaps we could test for both then? 

```js
options.environment === 'test' && options.environment === 'development'

```

Reply:
It's a complex topic for `http://localhost:4200/tests` and server middleware. The tests are running under test environment but the middleware is running under the environment used to start the server. So if we want to make sure that the nonce is only present in headers for testing we would end up with a check like this:

```js
options.environment === 'test' || req.path === '/tests'
```

To be honest I haven't checked if `ember test --server` runs middleware with `test` environment but I expect it to do so.

But I don't see a big issue in having that nonce present always in request headers. They don't influence build result but only used for development server (`ember serve`).

Reply:
Have done some quick tests and it seems like `serverMiddleware` hook is not executed if running `ember test --server`. CSP headers are not present at all.

Reply:
@jelhan Got it, makes sense! I know the test server is only running under development and won't affect builds. I'm just thinking that making that clear in the if-statement will make it easier to reason about the code.

But you could also add a comment instead, something like:

```js
// the local server will never run for production builds, so no danger in adding the nonce all the time
```

*(I still would have gone with an if-statement, but I don't have a strong opinion on this)*

Reply:
:+1: